packages/client/src/admin/Images/ImageAdd.tsx

```diff
@@ -0,0 +1,115 @@
+import React from 'react';
+import { useHistory } from 'react-router-dom';
+import { compose } from 'recompose';
+import { graphql } from 'react-apollo';
+import { omit } from 'lodash';
+import { notification } from 'antd';
+
+import { useRoutePrefix } from 'hooks/useRoutePrefix';
+
+import { ImagesLayout } from './Layout';
+import { ImageForm, ImageFormState } from './ImageForm';
+import { CreateImageMutation } from './images.graphql';
+import { errorHandler } from 'utils/errorHandler';
+
+interface Props {
+  createImageMutation: ({
+    variables,
+  }: {
+    variables: {
+      data: Partial<ImageFormState>;
+    };
+  }) => Promise<{ data: { createImage: { id: string } } }>;
+}
+
+function _ImageAdd({ createImageMutation }: Props) {
+  const history = useHistory();
+  const { appPrefix } = useRoutePrefix();
+
+  async function onSubmit({
+    isBuildByCustomImage,
+    ...restData
+  }: ImageFormState) {
+    // When creating an image, omit the `groups.disconnect` field
+    let formData = omit(
+      restData,
+      'groups.disconnect',
+      'baseImage',
+      'apt',
+      'conda',
+      'pip'
+    );
+
+    if (isBuildByCustomImage) {
+      formData = {
+        ...omit(formData, [
+          'pullSecret', // just only display on the custom image
+          'useImagePullSecret',
+        ]),
+        imageSpec: {
+          baseImage: restData.imageSpec.baseImage,
+          pullSecret: restData.imageSpec.pullSecret,
+          packages: {
+            apt: restData.apt.split('\n'),
+            conda: restData.conda.split('\n'),
+            pip: restData.pip.split('\n'),
```

Comment:
`packages.(apt|conda|pip)` should omit empty

current result:
```yaml
  packages:
    apt:
    - curl
    conda:
    - ""
    pip:
    - ""
```

expected:
```yaml
  packages:
    apt:
    - curl
```

Reply:
Updated, when one of the fields is empty, will send `null` instead of an empty string.

Reply:
Updated.